javascript-modules-engine-java/src/main/java/org/jahia/modules/javascript/modules/engine/jsengine/JSFileSystem.java

```diff
@@ -0,0 +1,135 @@
+package org.jahia.modules.javascript.modules.engine.jsengine;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.AccessMode;
+import java.nio.file.DirectoryStream;
+import java.nio.file.DirectoryStream.Filter;
+import java.nio.file.LinkOption;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Map;
+import java.util.Set;
+
+import org.graalvm.polyglot.io.FileSystem;
+import org.osgi.framework.BundleContext;
+import org.apache.commons.compress.utils.SeekableInMemoryByteChannel;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * This class defines a virtual filesystem used to resolve `import statements`.
```

Comment:
Unfortunately, Markdown in Javadoc is only supported since Java 23 (https://openjdk.org/jeps/467) :/
There are a few javadocs in this file that should be updated with HTML tags:
```suggestion
 * This class defines a virtual filesystem used to resolve <code>import statements</code>`.
```

Reply:
damn, done in 96051be5